staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go

```diff
@@ -86,7 +88,7 @@ func DialURL(url *url.URL, transport http.RoundTripper) (net.Conn, error) {
 			}
 
 		} else {
-			// Dial
+			// Dial. Cannot use ctx
```

Comment:
is this a todo?

Reply:
How about using tls.DialWithDialer(dialer *net.Dialer, network, addr string, config *Config), then should first construct a dialer with Deadline?